persistence/provider/memory/offsetstore.go

```diff
@@ -0,0 +1,101 @@
+package memory
+
+import (
+	"context"
+
+	"github.com/dogmatiq/infix/eventstream"
+	"github.com/dogmatiq/infix/persistence/subsystem/offsetstore"
+)
+
+// SaveOffset persists the "next" offset to be consumed for a specific
+// application.
+func (t *transaction) SaveOffset(
+	ctx context.Context,
+	ak string,
+	c, n eventstream.Offset,
+) error {
+	if err := t.begin(ctx); err != nil {
+		return err
+	}
+
+	if t.offset.stageSave(&t.ds.db.offset, ak, c, n) {
+		return nil
+	}
+
+	return offsetstore.ErrConflict
+}
+
+// offsetStoreChangeSet contains modifications to the offset store that have
+// been performed within a transaction but not yet committed.
+type offsetStoreChangeSet struct {
+	items map[string]eventstream.Offset
+}
+
+// stageSave adds a "SaveOffset" operation to the change-set.
+//
+// It returns false if there is an OCC conflict.
+func (cs *offsetStoreChangeSet) stageSave(
+	db *offsetStoreDatabase,
+	ak string,
+	c, n eventstream.Offset,
+) bool {
+	// Get both the committed item, and the item as it appears with its changes
+	// staged in this change-set.
```

Comment:
```suggestion
	// Get both the committed offset, and the staged offset from this change-set.
```

Yeah I know, hyper pedantic again, but there's nothing called an "item" here, blah blah :)